### PR TITLE
Moving bin prot wire-protocol related crates under /protocol #77

### PR DESCRIPTION
**Pull request by polarbeargo**
**Summary of changes**
Changes introduced in this pull request:
- 
Now the updated workspace Cargo.toml after this refactor look like the following 
```
[workspace]
members = [
	"base",
	"consensus",
	"crypto",
	# All crates related to bin prot wire protocol
	"protocol/bin-prot",
	"protocol/wire-type",
	"protocol/bin_prot_checker",
	"protocol/wire-type-derive",
	"protocol/test-fixtures",
	"protocol/test-serialization",
]

```

**Reference issue to close (if applicable)**
 Include the issue reference this pull request is connected to  #77 
Closes 

Screenshot (234)
Screenshot (235)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-	"bin_prot_checker",
 	"consensus",
 	"crypto",
-	"bin-prot",
-	"wire-type",
-	"wire-type-derive",
-	"test-fixtures",
-	"test-serialization",
+	# All crates related to bin prot wire protocol
+	"protocol/bin-prot",
```
Comment (hanabi1224) [inline Cargo.toml:7]: I think wildcard should be sufficient here

**Review by hanabi1224 — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (creativcoder) [reply Cargo.toml:7]: we might also put non-crates such as docs

**Comment by creativcoder** [comment]
We can also put layouts directory within protocol as its scope is only to be used by bin-prot related crates.

File: Cargo.toml (hunk 1 above)
Comment (polarbeargo) [reply Cargo.toml:7]: How about using [SwaggerHub](https://support.smartbear.com/swaggerhub/docs/enterprise/release-notes.html)?

**Comment by polarbeargo** [comment]
After putting the layouts directory within the protocol folder, cargo build & cargo test successful.
Screenshot (236)
Screenshot (237)

**Review by creativcoder — APPROVED** [review]
That was fast! :confetti_ball:

**Comment by polarbeargo** [comment]
@creativcoder thank you for the encouragement! Very appreciate 😊!

**Review by willemolding — APPROVED** [review]
I think also at some stage we should move a bunch of the types implementing WireType to the protocol subdir as well. Many of these won't really be used by the core logic and just serve to encode how the deserialization protocol should work.

This looks great though for a start :+1:

**Pull request merged**